src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -120,15 +124,37 @@ class UserService(protected val userInfo: UserInfo, dataSource: DataSource, prot
 
   //imports the response from the above listUsers
   def importUsers(rawlsUserInfoList: RawlsUserInfoList): Future[PerRequestMessage] = asAdmin {
-    Future(dataSource.inTransaction() { txn =>
-      val users = rawlsUserInfoList.userInfoList
-      users.foreach(u => {
-        containerDAO.authDAO.saveUser(u.user, txn)
-        u.billingProjects.foreach(b =>
-          containerDAO.billingDAO.addUserToProject(u.user, containerDAO.billingDAO.loadProject(b, txn).get, txn)
-        )
+    dataSource.inFutureTransaction() { txn =>
+      Future { 
+        val userInfos = rawlsUserInfoList.userInfoList
+        userInfos.map { u =>
```

Comment:
Why did this change from a foreach to a map? You're not using the returned list.